source/Mlos.Python/mlos/Optimizers/OptimizationProblem.py

```diff
@@ -6,20 +6,53 @@
 from typing import List, Tuple
 
 import pandas as pd
+import numpy as np
 
-from mlos.Spaces import SimpleHypergrid, CategoricalDimension
+from mlos.Spaces import SimpleHypergrid, CategoricalDimension, Dimension
 
-Objective = namedtuple("Objective", ["name", "minimize"])
 
-def objective_to_dict(objective):
-    return {
-        "name": objective.name,
-        "minimize": objective.minimize
-    }
+class Objective:
+    def __init__(self, name: str, minimize: bool):
+        self.name = name
+        self.minimize = minimize
 
 
-def objective_from_dict(objective_dict):
-    return Objective(**objective_dict)
+class SeriesObjective(Objective):
+    def __init__(self, name: str, minimize: bool, target_series: np.array, series_modulation_dimension: Dimension):

```

Comment:
Comments? Also, target_series cannot be an np.array - it has to be a dataframe with one column for time, and another for the target value at that time instance

Reply:
Also, formatting :) Also, stuffing the target_series_df into the Objective, is an elegant solution. 

Some general recommendations:
1. Make both Objective and SeriesObjective a @dataclass 
2. Rename SeriesObjective to something a brand new user can understand, such as SeriesMatchingObjective, or even TimeSeriesMatchingObjective. Do we envision trying to fit any other series other than time series?

Reply:
Ah! This explains a great deal

I have been using series_modulation_dimension to store the time values, but you rather suggest to store it in the series itself... that is far smarter. I will do this tomorrow. 

> Do we envision trying to fit any other series other than time series? > No reason to lock ourselves in right now I think